test/extensions/filters/http/wasm/test_data/grpc_callout_cpp.cc

```diff
@@ -40,23 +40,23 @@ class CalloutResponseHandler : public GrpcCallHandler<google::protobuf::Value> {
  public:
   CalloutResponseHandler(RequestContext *context)
       : GrpcCallHandler<google::protobuf::Value>(context), context_(context) {}
-  void onCreateInitialMetadata() override {}
+
   void onSuccess(google::protobuf::Value &&response) override {
     logDebug(response.string_value());
 
     serviceContext()->incrementCalloutSuccesses();
 
     continueRequest();
   }
+
   void onFailure(GrpcStatus status,
                  std::unique_ptr<WasmData> error_message) override {
     logInfo(std::string("failure ") + std::to_string(static_cast<int>(status)) +
-            std::string(error_message->view()));
+            " " + std::string(error_message->view()));
```

Comment:
absl is somewhat complicated inside of the sandbox because bazel loads the dependencies and it doesn't load all of absl and moreover we would want to have the headers in a known place.

The net net is that mixer is using it but I didn't want to have 2 absl's in the envoy tree so the test cases do not use it.

Reply:
On this one I'm just following with the other test cases do.  I don't want to be a special flower.

But... I wonder if the intrinsics should have some sort of printf/format-style that doesn't execute the arguments if the configured log level is less verbose than the log statement?

Reply:
Discussed with Mandar.  That is a bit tricky since the log level can be changed dynamically and we would need to make a call out of the VM to get the level which is likely going to be more expensive then just making the logging call and blocking it on the other side.  There are potential ways to work around this to some extent, but no easy answer AFAICT.